### PR TITLE
organising _CoqProject

### PR DESCRIPTION
**Pull request by Alizter**
This was meant to be part of the Torus PR but since that one is making travis cry I will add this change separately whilst I work out what is wrong with the other.

I haven't bothered giving sections to subfolders but those are typically easier to navigate than the rest.

### CODE REVIEW

File: _CoqProject [hunk 1]
```diff
@@ -12,6 +16,11 @@ theories/Basics/Equivalences.v
 theories/Basics/Trunc.v
 theories/Basics/Decidable.v
 theories/Basics.v
+
+#
+#   Types_CoqProject _CoqProject _CoqProject
```
Comment (mikeshulman) [inline _CoqProject]: What?

Comment (Alizter) [reply _CoqProject]: Hmm I have no idea how this got in here. Perhaps I ran a rogue script at some point.

**Comment by Alizter** [comment]
@mikeshulman That's all fixed. The merge feature on github did something weird that is all.

**Comment by mikeshulman** [comment]
LGTM

**Comment by mattam82** [comment]
It was probably not meant to stop compiling and installing the contrib/ files was it? With this  `hoqc`/`hoqtop` now fail to find the `share/contrib` directory

**Comment by Alizter** [comment]
@mattam82 I don't think this has anything to do with this change. I would delete share/contrib and share/theories and reinstall after compiling and see if that changes anything.

**Comment by mattam82** [comment]
Looking at the changes of this PR, you can see that none of the files in contrib/ are built anymore (they’re commented out), so when calling hoqdep, or hoqc after installation, it uses an option -R .../contrib that makes it error fatally because the directory no longer exists.

**Comment by SkySkimmer** [comment]
Why list files in _coqproject instead of doing a find in the makefile? (considering hott doesn't use coq_makefile)

**Comment by mattam82** [comment]
For IDEs I think

**Comment by SkySkimmer** [comment]
I don't think ides look at the list of files

Gaëtan Gilbert

**Comment by JasonGross** [comment]
> Why list files in _coqproject instead of doing a find in the makefile? (considering hott doesn't use coq_makefile)

I am opposed to using `find` because it will pick up non-version-controlled files.  I'd be more okay with `git ls-files`, but we support building from a tar ball, not just git, and it seems slightly poor to have a different file selection mechanism depending on whether or not we use git.

**Comment by SkySkimmer** [comment]
Why is picking up non vc files a problem?

Gaëtan Gilbert

**Comment by mikeshulman** [comment]
Make should only be making files that are part of the library.  A user may have their own .v files in the directory that are not part of the library.

**Comment by Alizter** [comment]
@mattam82 My apologies, you are correct. I got confused since in #1066 I uncommented the contrib lines in the _CoqProject file, instead of doing it here. I should have been more careful.

**Pull request merged**